### PR TITLE
Put in quotes the top_dir and test_path

### LINKED ISSUE #211: stestr fails if top_dir or test_path contains space
### Issue description

I case of the `top_dir` or `test_path` in the `.stestr.conf` file contains a space the `stestr run` or `stestr list` command fails with exception:
> ImportError: Start directory is not importable:

#### Expected behavior and actual behavior

stestr should accept the paths with spaces.

#### Steps to reproduce the problem

Have a project in directory with space.

### System information

**stestr version (`stestr --version`):**
stestr 2.2.1.dev8

**Python release (`python --version`):**
Python 3.7.2

**pip packages (`pip freeze`):**
tempest 19.0.1.dev244
and 104 packages

### Additional information

Here is `.stestr.conf` file:
```
[DEFAULT]
test_path=/Users/svilgelm/Library/Mobile Documents/com~apple~CloudDocs/workspace/qa/tempest/tempest/test_discover
top_dir=/Users/svilgelm/Library/Mobile Documents/com~apple~CloudDocs/workspace/qa/tempest/tempest
group_regex=([^\.]*\.)*
```

Traceback:
```
Traceback (most recent call last):
  File "/usr/local/Cellar/python/3.7.2/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/local/Cellar/python/3.7.2/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/Users/svilgelm/.local/share/virtualenvs/qa-BDEBjO9C/lib/python3.7/site-packages/subunit/run.py", line 149, in <module>
    main()
  File "/Users/svilgelm/.local/share/virtualenvs/qa-BDEBjO9C/lib/python3.7/site-packages/subunit/run.py", line 145, in main
    stdout=stdout, exit=False)
  File "/Users/svilgelm/.local/share/virtualenvs/qa-BDEBjO9C/lib/python3.7/site-packages/testtools/run.py", line 171, in __init__
    self.parseArgs(argv)
  File "/Users/svilgelm/.local/share/virtualenvs/qa-BDEBjO9C/lib/python3.7/site-packages/unittest2/main.py", line 113, in parseArgs
    self._do_discovery(argv[2:])
  File "/Users/svilgelm/.local/share/virtualenvs/qa-BDEBjO9C/lib/python3.7/site-packages/testtools/run.py", line 211, in _do_discovery
    super(TestProgram, self)._do_discovery(argv, Loader=Loader)
  File "/Users/svilgelm/.local/share/virtualenvs/qa-BDEBjO9C/lib/python3.7/site-packages/unittest2/main.py", line 223, in _do_discovery
    self.test = loader.discover(self.start, self.pattern, self.top)
  File "/Users/svilgelm/.local/share/virtualenvs/qa-BDEBjO9C/lib/python3.7/site-packages/unittest2/loader.py", line 364, in discover
    raise ImportError('Start directory is not importable: %r' % start_dir)
ImportError: Start directory is not importable: '/Users/svilgelm/Library/Mobile'

=========================
Failures during discovery
=========================

================================================================================
```

### PR DESCRIPTION
**Pull request by SVilgelm**
In case of top_dir and test_path contains a space, stestr fails.
Putting these variables in quotes fixes the problem.

Fix #211

### CODE REVIEW

**Comment by mtreinish** [comment]
This looks fine to me, but can we add a unit tests that exercises this edge case? I'm slightly concerned that it will behave differently on windows (just because I'm less familiar with that environment) and having a unit test will make sure this doesn't regress things on any of the supported environments.

**Comment by SVilgelm** [comment]
I change the _check_get_run_command and the unit tests passed on windows, so everything should work, but I will see ho can I improve it

File: stestr/config_file.py [hunk 1]
```diff
@@ -105,8 +105,8 @@ def get_run_command(self, test_ids=None, regexes=None,
 
         python = sys.executable if sys.platform == 'win32' else \
             '${PYTHON:-%s}' % (sys.executable)
-        command = "%s -m subunit.run discover -t" \
-                  " %s %s $LISTOPT $IDOPTION" % (python, top_dir, test_path)
+        command = '%s -m subunit.run discover -t "%s" "%s" ' \
```
Comment (mtreinish) [inline stestr/config_file.py]: Instead of manually inserting quotes here could we just call `shlex.quote()` (although we'll have to use `shlex_quote` from `six.moves` because python 2 vs python 3) I think properly escaping the shell command we're building here is just a good thing to do.

Comment (mtreinish) [reply stestr/config_file.py]: So I just tested this locally and using `shlex_quote()` doesn't have the desired behavior and it just breaks the execution of stestr. So leaving this as just quoted strings is fine.

**Comment by mtreinish** [comment]
I still think it'd be nice to have a unit test specifically testing the case this is being added for (the test path with a space in a directory name), but it's not a big deal and we can add that at a later date.
